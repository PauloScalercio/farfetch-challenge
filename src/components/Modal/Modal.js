import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ModalUi from '@material-ui/core/Modal';

function getModalStyle() {
  const top = 50;
  const left = 50 ;

  return {
    top: `${top}%`,
    left: `${left}%`,
    transform: `translate(-${top}%, -${left}%)`,
  };
}

const useStyles = makeStyles(theme => ({
  paper: {
    position: 'absolute',
    width: 400,
    backgroundColor: theme.palette.background.paper,
    border: '2px solid #000',
    boxShadow: '#000',
    padding: theme.spacing(2, 4, 3),
    ...getModalStyle()
  },
}));

const Modal = (props) => {
  const classes = useStyles();

  return (
    <div>
      <ModalUi
        {...props}
      >
        <div className={classes.paper}>
          {props.children}
        </div>
      </ModalUi>
    </div>
  );
}
export default Modal;