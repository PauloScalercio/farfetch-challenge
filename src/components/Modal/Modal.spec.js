import React from 'react';

import Modal from './index';

describe('Atom - Modal', () => {
  it('should preserve structure', () => {
    const component =  <Modal />;
    expect(component).toMatchSnapshot();
  });
});
