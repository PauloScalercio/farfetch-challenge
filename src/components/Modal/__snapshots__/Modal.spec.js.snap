// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - Modal should preserve structure 1`] = `<Modal />`;
